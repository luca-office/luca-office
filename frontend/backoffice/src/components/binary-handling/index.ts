export * from "./binary-form-placeholder/binary-form-placeholder"
export * from "./binary-update-modal/binary-entry"
export * from "./binary-update-modal/binary-update-modal"
export * from "./detail-view-binary/detail-view-binary"
export * from "./image-upload-modal/image-upload-modal"
