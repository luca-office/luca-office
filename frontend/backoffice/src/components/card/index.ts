// special backoffice card elements
export * from "./creation-card"
export * from "./duration-info/card-duration-info"
export * from "./tabbed-card/__mocks__/card-tabs.mock"
export * from "./tabbed-card/tabbed-card"
