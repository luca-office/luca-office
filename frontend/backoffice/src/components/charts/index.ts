export * from "./donut-chart/donut-chart-with-legend"
