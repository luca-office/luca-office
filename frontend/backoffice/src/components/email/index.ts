export * from "./email-body-meta-entry/email-body-meta-entry"
export * from "./email-body-setting-selection/email-body-setting-selection"
export * from "./email-content/email-content"
export * from "./email-directories-list/email-directories-list"
export * from "./email-header/email-header"
export * from "./email-unread-indicator/email-unread-indicator"
