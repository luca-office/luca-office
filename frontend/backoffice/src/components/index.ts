export * from "./binary-handling"
export * from "./card"
export * from "./card-overview/card-overview"
export * from "./charts"
export * from "./creation-bar/creation-bar"
export * from "./creation-status/creation-status"
export * from "./editable-multiline-textarea/editable-multiline-textarea"
export * from "./editing-status-indicator/editing-status-indicator"
export * from "./email"
export * from "./information-entry/information-entry"
export * from "./inline-editable-header/inline-editable-header"
export * from "./inline-editable-textarea/inline-editable-textarea"
export * from "./markdown-text/markdown-text"
export * from "./markdown-text-field/markdown-text-field"
export * from "./markdown-text-update-modal/markdown-text-update-modal"
export * from "./meta-entry/meta-entry"
export * from "./meta-entry-tags/meta-entry-tags"
export * from "./overlay-edit-field/overlay-edit-field"
export * from "./rating-progress-card/rating-progress-card"
export * from "./rating-progress-card/utils"
export * from "./resort-modal/resort-modal"
export * from "./sub-header-detail/sub-header-detail"
export * from "./sub-header-detail/sub-header-detail-container"
export * from "./sub-header-filter/sub-header-filter"
export * from "./table-action-bar/table-action-bar"
export * from "./upload-file-modal/upload-file-modal"
