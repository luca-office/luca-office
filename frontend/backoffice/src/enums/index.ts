export * from "./app-mode"
export * from "./creation-status"
export * from "./email-status"
export * from "./email-update-key"
export * from "./entity-filter"
export * from "./questionnaire-detail-mode"
export * from "./scenario-update-keys"
export * from "./survey-timing"
export * from "./user-management"
