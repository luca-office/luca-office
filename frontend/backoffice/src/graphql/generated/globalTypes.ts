/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum MimeType {
  ApplicationPdf = "ApplicationPdf",
  ImageGif = "ImageGif",
  ImageJpeg = "ImageJpeg",
  ImagePng = "ImagePng",
  ImageSvg = "ImageSvg",
  Spreadsheet = "Spreadsheet",
  TextHtml = "TextHtml",
  VideoMp4 = "VideoMp4",
}

export enum Relevance {
  Irrelevant = "Irrelevant",
  PotentiallyHelpful = "PotentiallyHelpful",
  Required = "Required",
}

export interface DirectoryCreation {
  name: string;
  parentDirectoryId?: string | null;
  scenarioId?: string | null;
  sampleCompanyId?: string | null;
}

export interface DirectoryUpdate {
  name: string;
  parentDirectoryId?: string | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
