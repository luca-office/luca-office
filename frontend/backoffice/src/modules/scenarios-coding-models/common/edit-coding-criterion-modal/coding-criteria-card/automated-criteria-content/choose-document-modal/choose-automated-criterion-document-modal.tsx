import {css} from "@emotion/react"
import * as React from "react"
import {Modal, Overlay, SelectableCard} from "shared/components"
import {IconName} from "shared/enums"
import {OfficeTool} from "shared/graphql/generated/globalTypes"
import {spacingMedium, spacingTiny} from "shared/styles"
import {useLucaTranslation} from "shared/translations"

export interface ChooseDocumentModalProps {
  readonly disabled?: boolean
  readonly onDismiss: () => void
  readonly onConfirm: (documentType: OfficeTool) => void
  readonly defaultDocumentType: OfficeTool
  readonly isDocumentViewCriterion: boolean
  readonly sampleCompanyId?: UUID
}

export const AutomatedCriterionChooseDocumentModal: React.FC<ChooseDocumentModalProps> = ({
  disabled = false,
  onDismiss,
  onConfirm,
  defaultDocumentType,
  sampleCompanyId,
  isDocumentViewCriterion
}) => {
  const {t} = useLucaTranslation()

  const [selectedDocumentType, setSelectedDocumentType] = React.useState(defaultDocumentType) // only state, no extra container for it

  return (
    <Overlay>
      <Modal
        customStyles={styles.modal}
        customFooterStyles={styles.footer}
        onConfirm={() => onConfirm(selectedDocumentType)}
        confirmButtonDisabled={disabled}
        onDismiss={onDismiss}
        title={t("coding_models__automated_item_input_value_choose_document_modal_title")}
        preventDismissOnEscape={true}
        preventSubmitOnEnter={true}>
        <div css={styles.content}>
          <SelectableCard
            title={t("coding_models__automated_item_input_value_position_all_mails")}
            iconName={IconName.Email}
            text={t(
              isDocumentViewCriterion
                ? "coding_models__automated_item_document_view_choose_document_modal_mails"
                : "coding_models__automated_item_input_value_choose_document_modal_mails_text"
            )}
            selected={selectedDocumentType === OfficeTool.EmailClient}
            onClick={() => setSelectedDocumentType(OfficeTool.EmailClient)}
          />
          <SelectableCard
            title={t(
              isDocumentViewCriterion
                ? "scenario_details__settings_label_files_and_directories"
                : "scenario_details__settings_label_files_and_directories_only_spreadsheet"
            )}
            iconName={IconName.FolderStack}
            text={t(
              isDocumentViewCriterion
                ? "coding_models__automated_item_document_view_choose_document_modal_files_and_directories"
                : "coding_models__automated_item_input_value_choose_document_modal_files_and_directories_text"
            )}
            selected={
              selectedDocumentType === OfficeTool.FileBrowser || selectedDocumentType === OfficeTool.SpreadsheetEditor
            }
            onClick={() =>
              setSelectedDocumentType(isDocumentViewCriterion ? OfficeTool.FileBrowser : OfficeTool.SpreadsheetEditor)
            }
          />
          {isDocumentViewCriterion && (
            <SelectableCard
              title={t("reference_book__title")}
              iconName={IconName.Book}
              text={t("coding_models__automated_item_document_view_choose_document_modal_reference_books")}
              selected={selectedDocumentType === OfficeTool.ReferenceBookViewer}
              onClick={() => setSelectedDocumentType(OfficeTool.ReferenceBookViewer)}
            />
          )}
          {isDocumentViewCriterion ? (
            <SelectableCard
              title={t("erp__title")}
              iconName={IconName.DataSet}
              disabled={sampleCompanyId === undefined}
              text={t("coding_models__automated_item_document_view_choose_document_modal_erp")}
              selected={selectedDocumentType === OfficeTool.Erp}
              onClick={() => setSelectedDocumentType(OfficeTool.Erp)}
            />
          ) : (
            <SelectableCard
              title={t("coding_models__automated_item_input_value_position_notes")}
              iconName={IconName.Notes}
              text={t("coding_models__automated_item_input_value_choose_document_modal_notes_text")}
              selected={selectedDocumentType === OfficeTool.Notes}
              onClick={() => setSelectedDocumentType(OfficeTool.Notes)}
            />
          )}
        </div>
      </Modal>
    </Overlay>
  )
}

const styles = {
  modal: css({
    width: 900
  }),
  content: css({
    display: "grid",
    gridTemplateColumns: "1fr 1fr",
    gridGap: spacingMedium,
    padding: spacingTiny
  }),
  footer: css({
    marginTop: spacingMedium
  })
}
