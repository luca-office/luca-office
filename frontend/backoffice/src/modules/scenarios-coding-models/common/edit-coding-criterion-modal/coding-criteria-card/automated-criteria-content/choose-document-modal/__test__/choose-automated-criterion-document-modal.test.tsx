import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Modal, Overlay, SelectableCard} from "shared/components"
import {IconName} from "shared/enums"
import {OfficeTool} from "shared/graphql/generated/globalTypes"
import {AutomatedCriterionChooseDocumentModal} from "../choose-automated-criterion-document-modal"

describe("automated-criterion-choose-document-modal", () => {
  it("renders correctly", async () => {
    const component = create(
      <AutomatedCriterionChooseDocumentModal
        defaultDocumentType={OfficeTool.FileBrowser}
        isDocumentViewCriterion={false}
        onConfirm={jest.fn()}
        onDismiss={jest.fn()}
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure - no document view", () => {
    const component = shallow(
      <AutomatedCriterionChooseDocumentModal
        defaultDocumentType={OfficeTool.FileBrowser}
        isDocumentViewCriterion={false}
        onConfirm={jest.fn()}
        onDismiss={jest.fn()}
      />
    )

    expect(component.find(Modal)).toHaveLength(1)
    expect(component.find(Overlay)).toHaveLength(1)
    expect(component.find(SelectableCard)).toHaveLength(3)
  })
  it("has correct structure - document view", () => {
    const component = shallow(
      <AutomatedCriterionChooseDocumentModal
        defaultDocumentType={OfficeTool.FileBrowser}
        isDocumentViewCriterion={true}
        onConfirm={jest.fn()}
        onDismiss={jest.fn()}
      />
    )

    expect(component.find(Modal)).toHaveLength(1)
    expect(component.find(Overlay)).toHaveLength(1)
    expect(component.find(SelectableCard)).toHaveLength(4)
    expect(component.find(SelectableCard).last().prop("iconName")).toEqual(IconName.DataSet)
  })
  it("selects default Document Type", () => {
    const component = shallow(
      <AutomatedCriterionChooseDocumentModal
        defaultDocumentType={OfficeTool.EmailClient}
        isDocumentViewCriterion={true}
        onConfirm={jest.fn()}
        onDismiss={jest.fn()}
      />
    )

    expect(component.find(SelectableCard).first().prop("selected")).toBe(true)
  })
  it("handles confirmation action", () => {
    const confirm = jest.fn()
    const component = shallow(
      <AutomatedCriterionChooseDocumentModal
        defaultDocumentType={OfficeTool.EmailClient}
        isDocumentViewCriterion={true}
        onConfirm={confirm}
        onDismiss={jest.fn()}
      />
    )

    component.find(SelectableCard).first().simulate("click")
    component.find(Modal).dive().find(".confirm-button").simulate("click")

    expect(confirm).toBeCalledWith(OfficeTool.EmailClient)
  })
})
