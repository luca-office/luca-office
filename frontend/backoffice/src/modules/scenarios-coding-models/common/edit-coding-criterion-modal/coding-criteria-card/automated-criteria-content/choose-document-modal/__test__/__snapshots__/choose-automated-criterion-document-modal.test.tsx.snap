// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`automated-criterion-choose-document-modal renders correctly 1`] = `
<div
  className="css-1iw6hgd-Overlay"
>
  <div
    className="css-1iw6hgd-Overlay"
  >
    <div>
      <div
        className="css-1cw1kcy-Modal"
      >
        <header
          className="css-14sl779-Modal"
        >
          <div
            className="css-1f6p2vj-Heading"
          >
            coding_models__automated_item_input_value_choose_document_modal_title
          </div>
          <div
            className="css-12vbzj8-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[MockFunction]}
          />
        </header>
        <section
          className="css-1gs9x9w-Modal"
        >
          <div
            className="css-1ooi5i8-AutomatedCriterionChooseDocumentModal"
          >
            <div
              className="css-o2tml0-Card"
              onClick={[Function]}
            >
              <div
                className="card-header css-a0e080-CardHeader"
              >
                <div
                  className="css-1cygpbo-SelectableCard"
                >
                  <div
                    className="css-17tkprm-SelectableCard"
                  >
                    <div
                      className="css-1lrzij5-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                    <div
                      className="css-1f6p2vj-Heading"
                    >
                      coding_models__automated_item_input_value_position_all_mails
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="css-hpovgn-Text"
              >
                coding_models__automated_item_input_value_choose_document_modal_mails_text
              </div>
            </div>
            <div
              className="css-1pgmn6a-Card"
              onClick={[Function]}
            >
              <div
                className="card-header css-a0e080-CardHeader"
              >
                <div
                  className="css-1cygpbo-SelectableCard"
                >
                  <div
                    className="css-17tkprm-SelectableCard"
                  >
                    <div
                      className="css-1lrzij5-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                    <div
                      className="css-1f6p2vj-Heading"
                    >
                      scenario_details__settings_label_files_and_directories_only_spreadsheet
                    </div>
                  </div>
                  <div
                    className="css-l0jg45-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="css-hpovgn-Text"
              >
                coding_models__automated_item_input_value_choose_document_modal_files_and_directories_text
              </div>
            </div>
            <div
              className="css-o2tml0-Card"
              onClick={[Function]}
            >
              <div
                className="card-header css-a0e080-CardHeader"
              >
                <div
                  className="css-1cygpbo-SelectableCard"
                >
                  <div
                    className="css-17tkprm-SelectableCard"
                  >
                    <div
                      className="css-1lrzij5-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                    <div
                      className="css-1f6p2vj-Heading"
                    >
                      coding_models__automated_item_input_value_position_notes
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="css-hpovgn-Text"
              >
                coding_models__automated_item_input_value_choose_document_modal_notes_text
              </div>
            </div>
          </div>
        </section>
        <footer
          className="css-159yf8u-Modal"
        >
          <button
            className="cancel-button css-a1i4d5-Button"
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="css-95j2hm-Button"
            >
              cancel_button
            </div>
          </button>
          <button
            className="confirm-button css-1ex7tyh-Button"
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="css-95j2hm-Button"
            >
              confirm_button
            </div>
          </button>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
